config/render.go

```diff
@@ -0,0 +1,75 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+// RenderApp encapsulates the rendered configuration file and the input and
+// output paths.
+type RenderApp struct {
+	renderFlag     string
+	renderedConfig []byte
+}
+
+// EmptyApp creates an empty application
+func EmptyApp() *RenderApp {
+	app := &RenderApp{}
+	return app
+}
+
+// NewRenderApp creates a new Config Rendering application
+func NewRenderApp(configFlag string, renderFlag string) (*RenderApp, error) {
```

Comment:
I would much rather see a `Render` method on `Config` than have this logic in a new interface around `App`. I see why you've created the new interface to begin with but it should delegate all the behavior to `Config`. The approach here ends up duplicating a bunch of code and that's going to be a source of bugs (like for example in this case we've left out all the template filter functions). Instead we should be calling `ParseConfig` to get a fully-parsed configuration that is identical to the one that the main application uses, and then making a `Render` call to that to dump it.

I suspect we should have this `Render` method dump a string (i.e. should have a signature like `func (c *Config) Render() string`) and then have the caller decide what to do with the string.

I'm also wondering whether we need to have a new interface on `App` at all. For the `version` flag we just did this right in the `LoadApp` and exited early: https://github.com/joyent/containerpilot/pull/271/files#diff-639ed475181e0558956e92333cd5e9c0R81